### DISCUSSION TITLE
error in borg after power outage

### DISCUSSION THREAD

**Discussion by alansenairj** [borgbackup/borg#8064] [Q&A] [answered]
Hello guys. I am new using borg and I don't know what to do to restart my backup. 

I am using Vorta to create a backup. When my backup was working I have a power outage. 

Now I am getting some python errors and i don't know how to solve it. Let me describe better. 

log: 

❯ borg check /mnt/sdd1/backup_home
Local Exception
Traceback (most recent call last):
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 996, in check
    transaction_id = self.get_transaction_id()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 415, in get_transaction_id
    self.check_transaction()
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 396, in check_transaction
    segments_transaction_id = self.io.get_segments_transaction_id()
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 1379, in get_segments_transaction_id
    if self.is_committed_segment(segment):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 1413, in is_committed_segment
    if fd.read(self.header_fmt.size) != self.COMMIT:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 5] Input/output error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.12/site-packages/borg/archiver.py", line 5343, in main
    exit_code = archiver.run(args)
                ^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/archiver.py", line 5263, in run
    return set_ec(func(args))
                  ^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/archiver.py", line 189, in wrapper
    return method(self, args, repository=repository, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/archiver.py", line 349, in do_check
    if not repository.check(repair=args.repair, save_space=args.save_space, max_duration=args.max_duration):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 1000, in check
    transaction_id = self.io.get_segments_transaction_id()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 1379, in get_segments_transaction_id
    if self.is_committed_segment(segment):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.12/site-packages/borg/repository.py", line 1413, in is_committed_segment
    if fd.read(self.header_fmt.size) != self.COMMIT:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 5] Input/output error

Platform: Linux z390 6.6.13-200.fc39.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Jan 20 18:03:28 UTC 2024 x86_64
Linux: Unknown Linux  
Borg: 1.2.7  Python: CPython 3.12.1 msgpack: 1.0.5 fuse: llfuse 1.5.0 [pyfuse3,llfuse]
PID: 12124  CWD: /home/alan
sys.argv: ['/usr/bin/borg', 'check', '/mnt/sdd1/backup_home']
SSH_ORIGINAL_COMMAND: None

![image](https://github.com/borgbackup/borg/assets/20565821/de9dd0ce-bed6-4ac1-adc7-04b6ee46a9e2)

I did a reinstall: 
sudo dnf reinstall borgbackup.x86_64

About borg I was working. Now It not finishing a command without crash.
I read docs but I don't know how to deal with it.

**Comment by alansenairj** [comment]
![image](https://github.com/borgbackup/borg/assets/20565821/9aacd29f-7d98-4c6c-809f-4d626fb00a27)

![image](https://github.com/borgbackup/borg/assets/20565821/59cdee3d-ddc6-469f-960a-a3d73deecdb1)

**Comment by ThomasWaldmann** [comment]
I/O Error means it is an error below borg, either filesystem corrupted or bad storage media.

So run `fsck` on that filesystem, check the disk with `smartctl -t long`.

**Reply by alansenairj** [reply]
Hi! Thanks for your replay. 

I did ntfs checking. No errors. And no smart error. 

root@z390:/home/alan# ntfsfix /dev/sde1
Mounting volume... OK
Processing of $MFT and $MFTMirr completed successfully.
Checking the alternate boot sector... OK
NTFS volume version is 3.1.
NTFS partition /dev/sde1 was processed successfully.


![image](https://github.com/borgbackup/borg/assets/20565821/3a2d90d5-e998-4038-bb2b-e1670d823ad7)

Does it works well using same disk? I mean, use a repo for borg at same disk for mounted rclone drive? 

/dev/sde1          932G   17G  916G   2% /mnt/sdd1
csi:               1.0T  988M  1.0T   1% /mnt/sdd1/rclone_csi

**Reply by ThomasWaldmann** [reply]
Check the smart attributes, esp. reallocated and pending sectors (raw value).

**Comment by alansenairj** [comment]
I think I found one error. Joplin file structure creates too big filenames. It is syncing to my Onedrive at Rclone mounted drive. I need to enter in Windows and check disk from there to fix some errors. 

I will try to stop with joplin syncing for a while to test it.

**Comment by alansenairj** [comment, marked as answer]
I think I got it working. I will reproduce steps. 

1 - Removed entire folder
2 - Checked disk at Windows
3 - Stoped Joplin syncing
4 - filtered .cache from home backup
5 - create a new repo with other name

Now it is working. 
![image](https://github.com/borgbackup/borg/assets/20565821/0d9abcab-5b8c-43a4-a558-3186cc881b20)

Thanxx for any help

**Comment by ThomasWaldmann** [comment]
I am not familiar with "Joplin" or what you are doing there exactly.

But great if you could fix it. :-)
